frontend/src/metabase/lib/formatting/numbers.tsx

```diff
@@ -1,253 +1,51 @@
-    currency: options.currency,
-    currencyDisplay: options.currency_style,
-    // always use grouping separators, but we may replace/remove them depending on number_separators option
-    useGrouping: true,
-    minimumIntegerDigits: options.minimumIntegerDigits,
-    minimumFractionDigits: options.minimumFractionDigits,
-    maximumFractionDigits: options.maximumFractionDigits,
-    minimumSignificantDigits: options.minimumSignificantDigits,
-    maximumSignificantDigits: options.maximumSignificantDigits,
-  }) as any;
-}
-
-function formatNumberCompact(value: number, options: FormatNumberOptionsType) {
-  if (options.number_style === "percent") {
-    return formatNumberCompactWithoutOptions(value * 100) + "%";
-  }
-  if (options.number_style === "currency") {
-    try {
-      const nf = numberFormatterForOptions({
-        ...options,
-        ...COMPACT_CURRENCY_OPTIONS,
-      });
-
-      if (Math.abs(value) < DISPLAY_COMPACT_DECIMALS_CUTOFF) {
-        return nf.format(value);
-      }
-      const { value: currency } = nf
-        .formatToParts(value)
-        .find((p: any) => p.type === "currency");
-      return currency + formatNumberCompactWithoutOptions(value);
-    } catch (e) {
-      // Intl.NumberFormat failed, so we fall back to a non-currency number
-      return formatNumberCompactWithoutOptions(value);
-    }
-  }
-  if (options.number_style === "scientific") {
-    return formatNumberScientific(value, {
-      ...options,
-      // unsetting maximumFractionDigits prevents truncation of small numbers
-      maximumFractionDigits: undefined,
-      minimumFractionDigits: 1,
-    });
-  }
-  return formatNumberCompactWithoutOptions(value);
-}
-
-function formatNumberCompactWithoutOptions(value: number) {
-  if (value === 0) {
-    // 0 => 0
-    return "0";
-  } else if (Math.abs(value) < DISPLAY_COMPACT_DECIMALS_CUTOFF) {
-    // 0.1 => 0.1
-    return PRECISION_NUMBER_FORMATTER(value).replace(/\.?0+$/, "");
-  } else {
-    // 1 => 1
-    // 1000 => 1K
-    return Humanize.compactInteger(Math.round(value), 1);
-  }
-}
-
-// replaces the decimal and grouping separators with those specified by a NumberSeparators option
-function replaceNumberSeparators(formatted: any, separators: any) {
-  const [decimalSeparator, groupingSeparator] = (separators || ".,").split("");
-
-  const separatorMap = {
-    ",": groupingSeparator || "",
-    ".": decimalSeparator,
-  };
-
-  return formatted.replace(
-    /,|\./g,
-    (separator: "." | ",") => separatorMap[separator],
-  );
-}
-
-function formatNumberScientific(
-  value: number,
-  options: FormatNumberOptionsType,
 ) {
-  if (options.maximumFractionDigits) {
-    value = d3.round(value, options.maximumFractionDigits);
-  }
-  const exp = replaceNumberSeparators(
-    value.toExponential(options.minimumFractionDigits),
-    options?.number_separators,
-  );
-  if (options.jsx) {
-    const [m, n] = exp.split("e");
+  let formatted = format_number(number, kebab_keys(options));
```

Comment:
Doesn't seem right that we need to call `kebab_keys` before passing something to a cljs function. Could this be handled internally?